synapse/handlers/oidc_handler.py

```diff
@@ -883,6 +883,12 @@ async def oidc_response_to_user_attributes(failures):
                     userinfo, token, failures
                 )
             else:
+                # If the mapping provider does not support processing failures,
+                # do not continually generate the same Matrix ID since this will
+                # likely continue to fail.
+                if failures:
+                    raise RuntimeError("Mapping provider does not support de-duplicating Matrix IDs")
```

Comment:
Shouldn't this be a `SynapeError`? Otherwise this will return a 500 and log a stack trace.

Reply:
This gets caught and turned into a `MappingException`:

https://github.com/matrix-org/synapse/pull/8801/files#diff-05be763854c3b4a72fd8c85ffb5a6b3a10da786f46c53ca4b64d1f150d8fc992R176-R181

Reply:
I clarified this in the comment: 18fcaee858b113adf443db6b12e16f22b6fd3289